WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsFragment.kt

```diff
@@ -24,33 +28,78 @@ class ReaderInterestsFragment : Fragment(R.layout.reader_interests_fragment_layo
         super.onViewCreated(view, savedInstanceState)
+        initDoneButton()
         initViewModel()
     }
 
+    private fun initDoneButton() {
+        done_button.setOnClickListener {
+            viewModel.onDoneButtonClick()
+        }
+    }
+
     private fun initViewModel() {
-        viewModel = ViewModelProviders.of(this, viewModelFactory).get(ReaderInterestsViewModel::class.java)
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
+            .get(ReaderInterestsViewModel::class.java)
         startObserving()
     }
 
     private fun startObserving() {
         viewModel.uiState.observe(viewLifecycleOwner, Observer { uiState ->
             updateInterests(uiState.interestsUiState)
+            updateDoneButton(uiState.doneButtonUiState)
+        })
+
+        viewModel.navigateToDiscover.observe(viewLifecycleOwner, Observer { event ->
+            event?.getContentIfNotHandled()?.let {
+                navigateToDiscover()
+            }
         })
 
         viewModel.start()
     }
 
+    private fun updateDoneButton(doneButtonUiState: DoneButtonUiState) {
+        with(done_button) {
+            isEnabled = doneButtonUiState.enabled
+            text = getString(doneButtonUiState.titleRes)
+        }
+        uiHelpers.updateVisibility(done_button, doneButtonUiState.visible)
+    }
+
     private fun updateInterests(interestsUiState: List<InterestUiState>) {
         interestsUiState.forEachIndexed { index, interestTagUiState ->
-            val chip = layoutInflater.inflate(R.layout.reader_interest_filter_chip, interests_chip_group, false) as Chip
+            var chip = interests_chip_group.findViewWithTag<Chip>(interestTagUiState.title)
+            if (chip == null) {
+                chip = layoutInflater.inflate(
+                    R.layout.reader_interest_filter_chip,
+                    interests_chip_group,
+                    false
+                ) as Chip
+                with(chip) {
+                    layoutDirection = View.LAYOUT_DIRECTION_LOCALE
+                    tag = interestTagUiState.title
+                    setOnCheckedChangeListener { compoundButton, isChecked ->
+                        if (compoundButton.isPressed) {
+                            viewModel.onInterestAtIndexToggled(index, isChecked)
+                        }
+                    }
+                    interests_chip_group.addView(chip)
+                }
+            }
             with(chip) {
-                layoutDirection = View.LAYOUT_DIRECTION_LOCALE
                 text = interestTagUiState.title
-                setOnCheckedChangeListener { _, _ -> // TODO: Pass index to VM
-                }
+                isChecked = interestTagUiState.isChecked
             }
-
-            interests_chip_group.addView(chip)
         }
     }
+
+    private fun navigateToDiscover() {
+        val fragmentTransaction = parentFragmentManager.beginTransaction()
```

Comment:
We might want to consider using the ActivityLauncher and letting the parent handle the navigation. It feels like it's not responsibility of this fragment to replace it with "ReaderFragment". Wdyt?

Reply:
Can I do it in an upcoming PR? I need to figure how to replace a fragment in the `WPMainNavigationView` and don't want to block this PR because of it.

Reply:
We might want to consider replacing it in the ReaderFragment - I'm not sure. Either case, definitely a good idea to do this in a separate PR.